File: Service/src/main/java/com/github/spb/tget/demo/dao/ContactDao.java
package com.github.spb.tget.demo.dao;

import com.github.spb.tget.demo.data.ClientEntity;
import com.github.spb.tget.demo.data.ContactInformation;
import com.github.spb.tget.demo.repository.Repository;
import com.github.spb.tget.demo.repository.RepositoryFactory;

import java.util.List;

public class ContactDao {

    private Repository contactInformationRepository;
    private String repoType = System.getProperty("repoType");

    public ContactDao() {
        contactInformationRepository = RepositoryFactory.Companion.getContactInformationRepositoryByType(repoType);
    }

    public List<ContactInformation> getAllContacts() {
        return contactInformationRepository.getItems();
    }

    public void addContacts(List<ContactInformation> contacts, ClientEntity client) {
        contacts.forEach(c -> {
            c.setClient(client);
            contactInformationRepository.addItem(c);
        });
    }
}

File: Service/src/main/java/com/github/spb/tget/demo/dao/ContactDao.kt
package com.github.spb.tget.demo.dao

import com.github.spb.tget.demo.data.ClientEntity
import com.github.spb.tget.demo.data.ContactEntity
import com.github.spb.tget.demo.repository.Repository
import com.github.spb.tget.demo.repository.RepositoryFactory

class ContactDao {

    private val contactInformationRepository: Repository<ContactEntity>
    private val repoType = System.getProperty("repoType")

    init {
        contactInformationRepository = RepositoryFactory.getContactInformationRepositoryByType(repoType)
    }

    fun addContacts(contacts: List<ContactEntity>, client: ClientEntity) {
        contacts.forEach { c ->
            c.client = client
            contactInformationRepository.addItem(c)
        }
    }

    fun getAllContacts(): List<ContactEntity> {
        return contactInformationRepository.getItems()
    }
}
